### DISCUSSION TITLE
Remove unused JavaScript to reduce bytes consumed by network activity

### DISCUSSION THREAD

**Discussion by willvlad** [vercel/next.js#16519] [Help]
Pagespeed insights from Google is telling me to remove unused JS, which adds 4.5 secs to my code.

I've already added 
import dynamic from 'next/dynamic';

const DynamicHeroComponentWithCustomLoading = dynamic(
  () => import('../components/IndexPageComponents/HeroIndexPage'),
  { loading: () => Loading }
)

but it's still showing Time to Interactive 9.3 s! And this is just with a basic one small index page with a single webp image of 46kb...

Does anyone know if the commons JS be trimmed somehow else in an automatic way? 

3 secs are apparently taken by the following:
Category | Time Spent
-- | --
Script Evaluation | 1,842 ms
Other | 497 ms
Script Parsing & Compilation | 249 ms
Style & Layout | 191 ms
Rendering | 101 ms
Parse HTML & CSS | 93 ms
Garbage Collection | 38 ms

Any ideas would be welcome!

**Comment by jamesmosier** [comment]
[See this Discussion](https://github.com/vercel/next.js/discussions/13682#discussioncomment-21797) which describes what Lighthouse does but most importantly points out:

> It is somewhat inevitable that some subset of JavaScript will be unused when using someone's library

As far as your specific application, it is hard to say without seeing the code. Reducing dependencies is one way to cut down on bundle size.

**Comment by dpatel4-lat** [comment]
I think all dynamic import components are still adding to the commons.js size even if they are not used on the page. Causing this issue.

**Reply by ike666666** [reply]
Nextjs webpack packaging its own problems, no solution was found
